pkg/services/entityevents/service.go

```diff
@@ -0,0 +1,115 @@
+package entityevents
+
+import (
+	"context"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/setting"
+)
+
+type EntityEventType string
+
+const (
+	EntityEventTypeDelete EntityEventType = "delete"
+	EntityEventTypeCreate EntityEventType = "create"
+	EntityEventTypeUpdate EntityEventType = "update"
+)
+
+type EntityEvent struct {
+	Id        int64
+	EventType EntityEventType
+	Grn       string
+	Created   time.Time
+}
+
+type SaveActionCmd struct {
+	Grn       string
+	EventType EntityEventType
+}
+
+type Service interface {
+	Save(ctx context.Context, cmd SaveActionCmd) error
+	GetLast(ctx context.Context) (*EntityEvent, error)
+	GetAllAfter(ctx context.Context, id int64) ([]*EntityEvent, error)
+
+	Run(ctx context.Context) error
+
+	deleteEventsOlderThan(ctx context.Context, duration time.Duration) error
+}
```

Comment:
```suggestion
/** EntityNotificationService is a temporary solution to support change notifications in an HA setup
* With this service each system can query for any events that have happened since a fixed time
*/
type EntityNotificationService interface {
	SaveEvent(ctx context.Context, cmd SaveActionCmd) error
	GetLastEvent(ctx context.Context) (*EntityEvent, error)
	GetAllEventsAfter(ctx context.Context, id int64) ([]*EntityEvent, error)

	Run(ctx context.Context) error

	deleteEventsOlderThan(ctx context.Context, duration time.Duration) error
}
```

I don't think we want a new package for this -- can/should it live in the `pkg/services/entitystore/...` package?  It should really should be internal to the entity store, but we may need to expose it so that the current "save dashboard" code could also push to it